packages/block-editor/src/layouts/flow.js

```diff
@@ -126,10 +127,17 @@ export default {
 			! shouldSkipSerialization( blockName, 'spacing', 'blockGap' )
 				? blockGapStyleValue?.top
 				: 'var( --wp--style--block-gap )';
+		// Using important here for the padding to override the inline padding that could be potentially
+		// applied using the custom padding control before the layout inheritance is applied.
 
 		let output =
 			!! contentSize || !! wideSize
 				? `
+					${ appendSelectors( selector ) } {
+						padding: ${ padding?.top || 0 } ${ padding?.right || 0 } ${
+						padding?.bottom || 0
+				  } ${ padding?.left || 0 } !important;
```

Comment:
I don't think this is the best way to go about it. If user sets a padding, we should respect that, and work with its value to add negative margins to full-width items. This is making it impossible to have a Group with both layout (needed for e.g. full width images) and padding, which is not ideal, especially if background color is set:

<img width="889" alt="Screen Shot 2022-04-08 at 5 19 53 pm" src="https://user-images.githubusercontent.com/8096000/162384741-8047a631-ab32-4d74-830a-def6aaaf114c.png">

Reply:
> This is making it impossible to have a Group with both layout (needed for e.g. full width images) and padding

Yes, I found that a bit challenging in testing, too, that toggling "Inherit default layout" hid the padding controls, which felt unexpected.

Reply:
If we keep the padding visible when we inherit a layout, we'd create another unexpected behavior though:

 - A user can set a padding and this one will be used
 - The user can remove the specific padding and the one from the inherited layout will take over (instead of no padding), we'd have to set `0` to reset the padding.

Not sure if that's confusing enough or if we can explain it to the user.